tests/b2share_unit_tests/files/test_b2share_storage.py

```diff
@@ -214,3 +216,121 @@ def test_embargoed_records_with_external_pids(app, test_users,
+            record_data = json.loads(
+                resp.get_data(as_text=True))
+            return record_data
+
+    def test_search_deposits(user):
+        with app.test_client() as client:
+            login_user(user, client)
+            search_deposits_url = url_for(
+                'b2share_records_rest.b2rec_list', drafts=1, size=100)
+            headers = [('Content-Type', 'application/json'),
+                       ('Accept', 'application/json')]
+            resp = client.get(
+                search_deposits_url,
+                headers=headers)
+            deposit_search_res = json.loads(
+                resp.get_data(as_text=True))
+            return deposit_search_res
+
+    def test_search_records(user):
+        with app.test_client() as client:
+            login_user(user, client)
+            search_records_url = url_for(
+                'b2share_records_rest.b2rec_list', size=100)
+            headers = [('Content-Type', 'application/json'),
+                       ('Accept', 'application/json')]
+            resp = client.get(
+                search_records_url,
+                headers=headers)
+            record_search_res = json.loads(
+                resp.get_data(as_text=True))
+            return record_search_res
+
+    with app.app_context():
+        deposit = Deposit.get_record(deposit_with_external_pids.deposit_id)
+        deposit_data = test_get_deposit(deposit.pid.pid_value,
+                                        test_users['deposits_creator'])
+        print(deposit_data)
+        # assert that the external_pids are visible when getting a deposit
+        assert_external_files(
+            deposit,
+            deposit_data['metadata']['external_pids'])
+
+    with app.app_context():
+        deposit.submit()
+    with app.app_context():
+        deposit.publish()
+        # here we need the object_uuid of the record or the record pid
+        # which means I should use a resolver to get the b2rec
+        # from the deposit pid
+        record_resolver = Resolver(
+            pid_type='b2rec',
+            object_type='rec',
+            getter=B2ShareRecord.get_record,
+        )
+        record_pid, record = record_resolver.resolve(deposit.pid.pid_value)
+        record_data = test_get_record(
+            record_pid.pid_value, test_users['deposits_creator'])
+        assert_external_files(record, record_data['metadata']['external_pids'])
+        record_data = test_get_record(
+            record_pid.pid_value, test_users['normal'])
+        assert_external_files(record, record_data['metadata']['external_pids'])
+
+    with app.app_context():
+        deposit_search_data = test_search_deposits(
+            test_users['deposits_creator'])
+        assert deposit_search_data['hits']['total'] == 1
+        assert 'external_pids' not in deposit_search_data[
+            'hits']['hits'][0]['metadata']
+
+    with app.app_context():
+        record_search_data = test_search_records(
```

Comment:
We could add a few comments explaining why the external_metadata is visible and why it is not.